pkg/controller/migmigration/hooks.go

```diff
@@ -0,0 +1,203 @@
+package migmigration
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	migapi "github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (t *Task) runHooks() (bool, error) {
```

Comment:
I used slightly different names since the create/status stuff is tightly coupled. prepareJob and ensureJob instead of ensureJobCreated and checkJobStatus